iocore/net/UnixNet.cc

```diff
@@ -450,18 +435,25 @@ NetHandler::mainNetEvent(int event, Event *e)
   ink_assert(trigger_event == e && (event == EVENT_INTERVAL || event == EVENT_POLL));
   (void)event;
   (void)e;
+  return this->waitForActivity(-1);
+}
+
+int
+NetHandler::waitForActivity(ink_hrtime timeout)
+{
   EventIO *epd = nullptr;
 
   NET_INCREMENT_DYN_STAT(net_handler_run_stat);
+  SCOPED_MUTEX_LOCK(lock, mutex, this->thread);
```

Comment:
It would be a spin lock issue here.

Reply:
Could you be more explicit? This should be the same locking as if the event had been dispatched.

Reply:
Can you please have a test with global session, since we always try to take nh's lock in the `acquireSession`.

Reply:
Yahoo! has been running this change internally with global session sharing for over two years.

Reply:
@SolidWallOfCode If you believe the `SCOPED_MUTEX_LOCK` is not the key change, in my opinion, we should using `TRY_MUTX_LOCK` as the `process_event` does.